core/psa_crypto.c

```diff
@@ -325,6 +325,12 @@ static int psa_is_dh_key_size_valid(size_t bits)
 
 psa_status_t mbedtls_to_psa_error(int ret)
 {
+    /* PSA error codes (except PSA_SUCCESS) exist in the space between
+     * PSA_ERROR_GENERIC_ERROR and PSA_OPERATION_INCOMPLETE, so if we are
+     * passed a PSA error code then just return it. */
+    if (ret <= PSA_ERROR_GENERIC_ERROR && ret >= PSA_OPERATION_INCOMPLETE) {
+        return (psa_status_t)ret;
+    }
```

Comment:
Using specific error names is fragile because we might add another status that's e.g. `PSA_OPERATION_INCOMPLETE - 1`.

All legacy error codes are either in the range -127..-2 or -32767..-1024. See [`error .h`](https://github.com/Mbed-TLS/mbedtls/blob/mbedtls-3.6.3/include/mbedtls/error.h#L25) up to 3.x. There was a gap between -1024 and -128 (0 as the high-level module ID) that never got used. Going forward, if we add new error values, they'll be in the unified error space from the start. So I think we can assume that only values in these legacy ranges need to be translated.

So I would suggest
```suggestion
    /* Only legacy error codes need to be translated.
     * Those are either a low-level error code (-127..-2)
     * or a high-level error code (<= -0x1000).
     */
    if (ret > -0x1000 && ret < -0x80) {
        return (psa_status_t)ret;
    }
```

Reply:
Makes sense, cheers!